src/java.compiler/share/classes/javax/lang/model/package-info.java

```diff
@@ -46,12 +46,96 @@
  * language constructs, like the {@linkplain javax.lang.model.element
  * element} representing {@code java.util.Set}, and the family of
  * {@linkplain javax.lang.model.type types} that may be associated
- * with an element, like the raw type {@code java.util.Set}, {@code
+ * with an element, like the raw type {@code java.util.Set},{@code
  * java.util.Set<String>}, and {@code java.util.Set<T>}.
  *
  * <p>Unless otherwise specified, methods in this package will throw
  * a {@code NullPointerException} if given a {@code null} argument.
  *
+ * <h2><a id=elementsAndTypes>Elements and Types</a></h2>
+ *
+ * <h3><a id=DefUse>Definitions and Uses</a></h3>
+ *
+ * In broad terms the {@link javax.lang.model.element element} package
+ * models the declarations, that is the <em>definitions</em>, of elements while
+ * the {@link javax.lang.model.type type} package models <em>uses</em>
+ * of types. In general, distinct uses can have individualized
+ * information separate from the information associated with the
+ * definition. In some sense, the information in the definition is
+ * shared by all the uses.
+
+ * <p>For example, consider the uses of {@code
+ * java.lang.String} in the string processing method {@code
+ * identityOrEmpty} below:
+ *
+ * {@snippet lang=java :
+ * // Return the argument if it is non-null and the empty string otherwise.
+ * public static @DefinitelyNotNull String identityOrEmpty(@MightBeNull String argument) {
+ *    ...
+ * }
+ * }
+ *
+ * The return type of the method is a {@code String} annotated with
+ * a {@code @DefinitelyNotNull} type annotation while the type of
+ * the parameter is a {@code String} annotated with a {@code
+ * @MightBeNull} type annotation. In a reflective API, since the set
```

Comment:
> In a reflective API,

Nit: re-reading that bit made me pause for a moment and doubt that javax.lang.model is a reflective API, which I hope it is. We could help the reader by rephrasing that more assertively "In a reflective API, such as this one," or dropping "In a reflective API" altogether.

Reply:
Will change the opening sentence to:

"Types and hierarchies of packages comprising a Java language model, a reflective API that models the declarations and types of the Java programming language."

Reply:
(The text was initially written when core reflection and javax.lang.model were both discussed.)